src/transport/NetworkProvisioning.cpp

```diff
@@ -124,6 +124,11 @@ CHIP_ERROR NetworkProvisioning::HandleNetworkProvisioningMessage(uint8_t msgType
     return err;
 }
 
+uint16_t NetworkProvisioning::EncodedStringSize(const char * str)
+{
+    return static_cast<uint16_t>(strlen(str) + sizeof(uint16_t));
```

Comment:
```suggestion
    return static_cast<uint16_t>(sizeof(uint16_t) + strlen(str));
```
Also suggest using size_t here, not uint16_t, or at least internally use size_t and ensure it fits within 16 bits